Carbon.Cryptography/Functions/Grant-CPrivateKeyPermission.ps1

```diff
@@ -0,0 +1,278 @@
+
+    Demonstrates how to have the permission granted returned by using the `-PassThru` switch. If running under Windows
+    PowerShell and the .NET framework uses the
+    [RSACryptoServiceProvider](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsacryptoserviceprovider)
+    or the
+    [DSACryptoServiceProvider](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.dsacryptoserviceprovider)
+    class to manage the private key, the function returns `System.Security.AccessRule.CryptoKeyAccessRule` objects.
+    Otherwise, it returns `System.Security.AccessRule.FileSystemAccessRule` objects.
+    #>
+    [Diagnostics.CodeAnalysis.SuppressMessage('PSShouldProcess', '')]
+    [CmdletBinding(SupportsShouldProcess)]
+    [OutputType([Security.AccessControl.AccessRule])]
+    param(
+        [Parameter(Mandatory)]
+        # The path on which the permissions should be granted. Must be the path to an X509 certificate, i.e. an item in
+        # the PowerShell `cert:` drive.
+        [String] $Path,
+
+        # The user or group name getting the permissions.
+        [Parameter(Mandatory)]
+        [String] $Identity,
+
+        # The permission to grant. The Windows UI only allows Read and FullControl access, so
+        # `Grant-CPrivateKeyPermission` also only allows `Read` and `FullControl` permissions.
+        [Parameter(Mandatory)]
+        [ValidateSet('Read', 'FullControl')]
+        [String] $Permission,
+
+        # The type of rule to grant, either `Allow` or `Deny`. The default is `Allow`, which will allow access to the
+        # item. The other option is `Deny`, which will deny access to the item.
+        [Security.AccessControl.AccessControlType] $Type = [Security.AccessControl.AccessControlType]::Allow,
+
+        # Removes all non-inherited permissions on the item.
+        [switch] $Clear,
+
+        # Returns an object representing the permission created or set on the `Path`. The returned object will have a
+        # `Path` propery added to it so it can be piped to any cmdlet that uses a path.
+        [switch] $PassThru,
+
+        # Grants permissions, even if they are already present.
+        [switch] $Force
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    if (-not $IsWindows)
+    {
+        Write-Error -Message 'Grant-CPrivateKeyPermission only supports Windows.' -ErrorAction $ErrorActionPreference
+        return
+    }
+
+    if (-not (Test-Path -Path $Path))
+    {
+        $msg = "Failed to grant permissions on ""${Path}"" certificate's private key because that certificate does " +
+               'not exist.'
+        Write-Error -Message $msg -ErrorAction $ErrorActionPreference
+        return
+    }
+
+    if( -not (Test-CIdentity -Name $Identity ) )

```

Comment:
Style: Inconsistent style on spacing for `if` condition, e.g. `if (condition)` and `if( condition )`